### ISSUE TITLE
unnamed group after re-registering - can't respond

### ISSUE THREAD

**Issue by benmayne** [signalapp/Signal-Android#7728]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When I re-registered on a new phone, my current groups are "unknown group". I can't see the members and I can't reply. 

This is a partial dupe of https://github.com/signalapp/Signal-Android/issues/2581 (closed in cleanup) and probably https://github.com/signalapp/Signal-iOS/issues/2257

Getting one of the other members of the group to update it in any way fixes the problem. Since fixing the first group, from the same user to a different group updates the group successfully, so I can't quite reproduce this. I'll update this if it happens again with a more recent debug message. 

From desktop the groups seem to still work, but I can't edit the group to fix it. 

### Steps to reproduce
- get new device and register same number
- receive message to preexisting group

**Actual result:** group is unnamed and you can't reply. 
**Expected result:** I can reply to the thread, see the members, or update the group

### Screenshots

![screenshot_20180423-193532](https://user-images.githubusercontent.com/3374110/39164660-83df2318-4735-11e8-820a-e344d6e53846.png)

### Device info

**Device:** Pixel 2
**Android version:** 8.1.0
**Signal version:** 4.19.1

### Link to debug log
https://debuglogs.org/c48d4591213cc3d2df566ea545d451c437e222f0602e77f810d95eb839abb62d

**Comment by Hormesis** [comment]
I have this problem as well. If any further debug information is needed, I'm happy to provide it.

**Comment by benmayne** [comment]
I have a plausible second occurrence. I sent a message from my desktop app to a group, but it comes up as "unknown group" on android. 

desktop debug: https://debuglogs.org/a22a5a9e6760e49c5d9fa37897afe698c8ecb8495bfd7b149952481ef5d809cc
android debug: https://debuglogs.org/980ed6babbfb81f69e71d8b5bd1caf2b2aebc32f3d70de0ccbd4b25697cdd792

![2018-06-11 20 34 16](https://user-images.githubusercontent.com/3374110/41268997-b125e994-6db7-11e8-8990-2cdfd75bbbe4.png)

![2018-06-11 20 33 59](https://user-images.githubusercontent.com/3374110/41269003-b49033be-6db7-11e8-92b6-09cf748f0a8d.png)

**Comment by Tursko** [comment]
> I have a plausible second occurrence. I sent a message from my desktop app to a group, but it comes up as "unknown group" on android.

I have the same problem here. I got a new phone which had no history of my messages. I got on the desktop app and messaged a group I was in and on my phone, it shows Unknown group with no members UNTIL another person messages. Once they message I get full reference to the chat on my phone within the same thread. 

https://debuglogs.org/0402d86401434e3f68fdf34e1def628e17897047cf9aa4657e64db25b6a3a2c3
